### PR TITLE
Add analytics tags for email alerts

### PR DESCRIPTION
**Pull request by austensen**
Adds all the Google Tag Manager (and Amplitude) events for our tracking in GA4. 
Also makes updates to accommodate the user `id` and `type` now sent from auth provider, which we pass as params in our analytics events. 
Corresponding auth-provider pr: https://github.com/JustFixNYC/auth-provider/pull/39

[sc-14433]

### CODE REVIEW

File: client/src/components/EmailAlertSignup.tsx [hunk 1]
```diff
@@ -51,10 +51,14 @@ const BuildingSubscribeWithoutI18n = (props: BuildingSubscribeProps) => {
   const showSubscribed =
     (justSubscribed && !!user?.verified) || !!user?.subscriptions?.find((s) => s.bbl === addr.bbl);
 
+  const eventUserParams = { user_id: user?.id, user_type: user?.type };
+
   const [isEmailResent, setIsEmailResent] = React.useState(false);
   const [showSubscriptionLimitModal, setShowSubscriptionLimitModal] = useState(false);
 
   const navigateToLogin = () => {
+    // logAmplitudeEvent("registerLoginViaBuilding");
```
Comment (kiwansim) [inline client/src/components/EmailAlertSignup.tsx:60]: do we want to remove the amplitude event logs altogether? if I recall correctly, we haven't been tracking the login related events on amplitude thus far anyway

File: client/src/components/Login.tsx [hunk 2]
```diff
@@ -85,6 +87,19 @@ const LoginWithoutI18n = (props: withI18nProps) => {
   const [invalidAuthError, setInvalidAuthError] = useState(false);
   const [existingUserError, setExistingUserError] = useState(false);
 
+  const eventParams = (user?: JustfixUser) => {
```
Comment (kiwansim) [inline client/src/components/Login.tsx:90]: this is super neat!

**Review by kiwansim — APPROVED** [review]
🏷️✅

File: client/src/components/EmailAlertSignup.tsx (hunk 1 above)
Comment (austensen) [reply client/src/components/EmailAlertSignup.tsx]: yeah lets just not bother with it - removed all the comments

**Review by kiwansim — APPROVED** [review]

**Pull request merged**